### PR TITLE
add job level affinity doc

### PR DESCRIPTION
**Pull request by sxy390**
Adding the functional design doc for Job level affinity.

### CODE REVIEW

File: docs/functionality/job-level-affinity/job-level-affinity-overview.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+   apiVersion: scheduling.orchestration.bytedance.com/v1alpha1
+   kind: PodGroup
+   metadata:
+     generation: 1
+     name: nginx
+   spec:
+     affinity:
+       podGroupAffinity:
+         preferred:
+         - topologyKey: tce.kubernetes.io/nettoripv6
```
Comment (yuey002) [inline docs/functionality/job-level-affinity/job-level-affinity-overview.md:69]: Is it ok to use 'tce' for open-source docs? Shall we change the key name to a more general one?

File: docs/functionality/job-level-affinity/job-level-affinity-overview.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    apiVersion: scheduling.orchestration.bytedance.com/v1alpha1
+    kind: PodGroup
+    metadata:
+      generation: 1
+      name: nginx
+    spec:
+      affinity:
+        podGroupAffinity:
+        preferred:
+        - topologyKey: tce.kubernetes.io/nettoripv6
```
Comment (yuey002) [inline docs/functionality/job-level-affinity/job-level-affinity-overview.md:117]: Same for the key name here.

File: docs/functionality/job-level-affinity/network_topology_diagram.png
Comment (yuey002) [inline docs/functionality/job-level-affinity/network_topology_diagram.png:1]: From the team: please use more general terms to replace BigPod, MiniPod, and Tor.

File: docs/functionality/job-level-affinity/job-level-affinity-overview.md (hunk 1 above)
Comment (sxy390) [reply docs/functionality/job-level-affinity/job-level-affinity-overview.md:69]: done

**Review by lintongj — APPROVED** [review]

**Comment by rookie0080** [comment]
LGTM

**Review by binacs — APPROVED** [review]

**Pull request merged**